lib/stdlib/src/sets.erl

```diff
@@ -206,10 +235,20 @@ union1(S1, []) -> S1.
       Set1 :: set(Element),
       Set2 :: set(Element),
       Set3 :: set(Element).
-intersection(S1, S2) when S1#set.size < S2#set.size ->
-    filter(fun (E) -> is_element(E, S2) end, S1);
-intersection(S1, S2) ->
-    filter(fun (E) -> is_element(E, S1) end, S2).
+intersection(#{}=S1, #{}=S2) ->
+    if
```

Comment:
I noticed that maps were slower here, can you try the implementation from PR #2797?

Reply:
My plan is to not change the `cerl_sets` implementation for now and once that PR is merged, we can just delete to maps:intersection. What do you think?

Reply:
Sure, I just figured I'd point it out since you mentioned benchmarks and this was the only operation that was slower. #2797 should take care of that :)

Reply:
Oh, I see. Perfect, thanks. :)

Reply:
Moved the suggestion to the PR description.